examples/fabric-admin/rpc/RpcServer.cpp

```diff
@@ -48,19 +48,27 @@ class FabricAdmin final : public rpc::FabricAdmin, public IcdManager::Delegate
     void OnCheckInCompleted(const chip::app::ICDClientInfo & clientInfo) override
     {
         chip::NodeId nodeId = clientInfo.peer_node.GetNodeId();
-        auto it             = mPendingKeepActiveTimesMs.find(nodeId);
-        VerifyOrReturn(it != mPendingKeepActiveTimesMs.end());
-        // TODO(#33221): We also need a mechanism here to drop KeepActive
-        // request if they were recieved over 60 mins ago.
-        uint32_t stayActiveDurationMs = it->second;
+        auto it             = mPendingCheckIn.find(nodeId);
+        VerifyOrReturn(it != mPendingCheckIn.end());
+
+        KeepActiveDataForCheckIn checkInData = it->second;
+        // Removed from pending map as check-in from this node has occured and we will handle the pending KeepActive
+        // request.
+        mPendingCheckIn.erase(nodeId);
+
+        auto timeNow = System::SystemClock().GetMonotonicTimestamp();
+        if (timeNow > checkInData.mRequestExpiryTimestamp)
```

Comment:
Why is this kept around until the checkin? It seems likely that if the device hasn't checked in by the expiry time that it just is never going to and you'll have lots of dead entries in this map.

Reply:
This design use the absolute timestamp to compare, if we want it self deleted, then we need to start a timer for each of them.

Reply:
You will have up to 1 entry for each node you are a bridge for. For clients this is trivial amounts of memory

ICD device can take hours to check-in, eventually we expect these to expire. If we are really concerned we can add clean up code later, but I don't think it is worth the development time.